test/pylib: fix keyspace_compaction method

lkshminarayanan opened the pull request:
The [keyspace_compaction](https://github.com/scylladb/scylladb/blob/39dd08837456d517b82121ac062703c6e2f73a11/test/pylib/rest_client.py#L330) method incorrectly appends the column family parameter to the URL using a regular string, `"?cf={table}"`, instead of an f-string `f"?cf={table}"`.

Due to this, the call - `keyspace_compaction(node_ip, keyspace='ks', table='cf')` generates this URL for the POST request : `/storage_service/keyspace_compaction/ks?cf={table}`, (i.e) the table parameter never gets substituted and the POST request fails trying to find a column family named `{table}`.

Fix this issue by passing the parameter to the POST request using a dictionary instead of appending it to the URL.

Fixes #20264 

Backport needed: This isn't a critical fix, but it's necessary for any test cases using this method to work properly if backported.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11043/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11043/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11043/)
#### Build Details:
- Duration: 2 hr 4 min
- Builder: spider2.cloudius-systems.com

avikivity commented:
I don't understand what this fixes. And if it fixes something that deserves backporting, there should be an issue.

bhalevy commented on test/pylib/rest_client.py:
```diff
@@ -329,10 +329,10 @@ async def keyspace_flush(self, node_ip: str, keyspace: str, table: Optional[str]
 
     async def keyspace_compaction(self, node_ip: str, keyspace: str, table: Optional[str] = None) -> None:
         """Compact the specified or all tables in the keyspace"""
-        url = f"/storage_service/keyspace_compaction/{keyspace}"
```
I'd keep the `url =` assignment and usage below
to reduce the code churn

lkshminarayanan replied:
Ack. Done.

lkshminarayanan commented:
> I don't understand what this fixes. And if it fixes something that deserves backporting, there should be an issue.

I have created an issue and updated the cover letter.

lkshminarayanan commented:
> > The keyspace_compaction method passes the table name parameter as a regular string instead of an f-string. Fix that by passing the parameter as a dictionary to the post method.
> > Backport needed: This isn't a critical fix, but it's necessary for any test cases using this method to work properly if backported.
> 
> What breaks exactly when passing the table as part of the url vs. using the `params` dictionary?

I have updated the cover letter with more info. The string is basically missing an `f` prefix. Due to that the URL gets encoded as `/storage_service/keyspace_compaction/ks?cf={table}`. We can fix it by just prepending the `f` to the string but using a dictionary is more convenient, so I replaced it with that.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11070/)
#### Build Details:
- Duration: 40 min
- Builder: spider5.cloudius-systems.com

bhalevy reviewed (approved):
lgtm, thanks for clarifying the issue the patch fixes

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11076/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11076/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/11076/)
#### Build Details:
- Duration: 2 hr 55 min
- Builder: spider6.cloudius-systems.com
